src/brsTypes/components/RoSGNode.ts

```diff
@@ -317,6 +318,33 @@ export class RoSGNode extends BrsComponent implements BrsValue, BrsIterable {
         return false;
     }
 
+    /**
+     * Calls the function specified on this node.
+     */
+    private callFunc = new Callable("callFunc", {
+        signature: {
+            args: [
+                new StdlibArgument("functionname", ValueKind.String),
+                new StdlibArgument("functionargs", ValueKind.Dynamic),
+            ],
+            returns: ValueKind.Dynamic,
+        },
+        impl: (interpreter: Interpreter, functionname: BrsString, functionargs: BrsType) => {
+            // We need to search the callee's environment for this function rather than the caller's.
+            let componentDef = interpreter.environment.nodeDefMap.get(this.nodeSubtype);
+            if (componentDef) {
+                return interpreter.inSubEnv(subInterpreter => {
+                    let functionToCall = subInterpreter.getCallableFunction(functionname.value);
+                    return functionToCall.call(subInterpreter, functionargs);
+                }, componentDef.environment);
+            } else {
+                throw new Error(
+                    `Error: Unable to find function "${functionname}" used in callFunc.`
+                );
+            }
```

Comment:
Can we return `invalid` here? That's what I got from RBI in all cases (even when function doesn't exist)

Reply:
Hmmm I think it should return whatever the function returns (so if the function returns `boolean`, then this call should return `boolean`, etc). That should handle the `invalid` case if the function doesn't return anything.

But if the function doesn't exist (like you called out in a different comment), then we should `console.error` and return `invalid` instead of throwing an error. I'll change that!

Reply:
Oh yeah, you're right. Those invalids were actually returned by the functions :homerdisappears: